### ISSUE TITLE
Inliner changes behavior in PENNANT

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#446]
On a fresh pull of Legion master, go to `language/examples` and run `regent.py pennant_sequential.rg pennant.tests/sedovsmall/sedovsmall.pnt -npieces 1 -fparallelize 0 -fcuda 0`. You should get (at the end of the simulation):

```
Initializing (t=0.0)...
Starting simulation (t=0.0)...
Elapsed time = 7.383700e-02
Running validate_output (t=0.1)...
Reading "pennant.tests/sedovsmall/sedovsmall.xy.std"...
Successfully validate output
```

If you go into `pennant_sequential.rg` and add `__demand(__inline)` to task `continue_simulation`, the output of the same command becomes:

```
Initializing (t=0.0)...
Starting simulation (t=0.0)...
cycle   10    sim time 2.856e-02    dt 2.675e-03    time 7.445e-04 (per iteration) 7.445e-03 (total)
cycle   20    sim time 6.344e-02    dt 5.119e-03    time 6.559e-03 (per iteration) 7.304e-02 (total)
cycle   30    sim time 1.385e-01    dt 9.467e-03    time 6.517e-03 (per iteration) 1.382e-01 (total)
cycle   40    sim time 2.642e-01    dt 1.651e-02    time 6.551e-03 (per iteration) 2.037e-01 (total)
cycle   50    sim time 4.668e-01    dt 2.871e-02    time 6.583e-03 (per iteration) 2.695e-01 (total)
cycle   60    sim time 7.694e-01    dt 3.185e-02    time 6.563e-03 (per iteration) 3.352e-01 (total)
assertion failed: sv negative
```

Here is the pretty-printer's output for the offending loop, before inlining:

```
    var $alfa : double = $conf.alfa
    var $cfl : double = $conf.cfl
    var $cflv : double = $conf.cflv
    var $cstop : int64 = $conf.cstop
    var $dtfac : double = $conf.dtfac
    var $dtinit : double = $conf.dtinit
    var $dtmax : double = $conf.dtmax
    var $gamma : double = $conf.gamma
    var $q1 : double = $conf.q1
    var $q2 : double = $conf.q2
    var $qgamma : double = $conf.qgamma
    var $ssmin : double = $conf.ssmin
    var $tstop : double = $conf.tstop
    var $uinitradial : double = $conf.uinitradial
    var $vfix : vec2 = vec2({x = 0, y = 0})
    var $enable : bool = $conf.enable
    var $interval : int32 = 10
    var $start_time#14039 : double = (legion_get_current_time_in_micros()/1000000)
    var $last_time : double = $start_time#14039
    var $time : future(double) = __future(0)
    var $cycle : int64 = int64(0)
    var $dt : future(double) = __future($dtmax)
    var $dthydro : future(double) = __future($dtmax)
    while __future_get_result(continue_simulation($cycle, $cstop, $time, $tstop)) do
```

and after inlining:

```
    var $alfa : double = $conf.alfa
    var $cfl : double = $conf.cfl
    var $cflv : double = $conf.cflv
    var $cstop : int64 = $conf.cstop
    var $dtfac : double = $conf.dtfac
    var $dtinit : double = $conf.dtinit
    var $dtmax : double = $conf.dtmax
    var $gamma : double = $conf.gamma
    var $q1 : double = $conf.q1
    var $q2 : double = $conf.q2
    var $qgamma : double = $conf.qgamma
    var $ssmin : double = $conf.ssmin
    var $tstop : double = $conf.tstop
    var $uinitradial : double = $conf.uinitradial
    var $vfix : vec2 = vec2({x = 0, y = 0})
    var $enable : bool = $conf.enable
    var $interval : int32 = 10
    var $start_time#14039 : double = (legion_get_current_time_in_micros()/1000000)
    var $last_time : double = $start_time#14039
    var $time : future(double) = __future(0)
    var $cycle : int64 = int64(0)
    var $dt : future(double) = __future($dtmax)
    var $dthydro : future(double) = __future($dtmax)
    var $cycle#14047 : int64 = $cycle
    var $cstop#14048 : int64 = $cstop
    var $time#14340 : future(double) = $time
    var $tstop#14050 : double = $tstop
    var $__continue_simulation_ret : future(bool) = (($cycle#14047<$cstop#14048) and ($time#14340<$tstop#14050))
    while __future_get_result($__continue_simulation_ret) do
```

**Comment by magnatelee** [comment]
This should be fixed as of f7452239.

**Comment by manopapad** [comment]
Fixed!
